### PR TITLE
Modifications to bootstrap Roslyn on CoreClr without exceptions

### PR DESCRIPTION
**Pull request by richardlford**
This change solves a number of issues that were resulting in
problems compiling Roslyn with Roslyn on the CoreClr.
I believe these problems were being masked by catching
exceptions. The motivation for the changes is to be able
to run Roslyn building Roslyn on CoreClr using the LLILC jit.
LLILC cannot currently produce code to catch exceptions
so it is important to avoid needless throwing of exceptions.

1. CoreClr was raising FileNotFound exceptions when trying to load the
following assemblies:

    ```
    System.Threading.Overlapped
    System.Threading.ThreadPool
    ```

    I added references to these in src/Compilers/CSharp/CscCore/project.json.

2. System.Runtime in the version previously referenced in project.son
had a reference to Internal.Uri. This is no longer the current module
(it has been replaced by System.Private.Uri). So I changed the
version number used in project.json from 22816 to 23109.

3. The path in src/Compilers/Core/CodeAnalysisRules.ruleset
was "..\..\Tools\Microsoft.CodeAnalysis.Toolset.Open\Rulesets\Roslyn.ruleset"
which is not the current location of Roslyn.ruleset. It now is in
"..\..\..\build\Rulesets\Roslyn.ruleset". The incorrect path
was causing an exception to be thrown.

4. On the CoreClr csc uses a NoOpAnalyzerAssemblyLoader which just
returns null when asked to get the assembly. This was causing a
null reference exception. I added a null check to just return
in this case.

### CODE REVIEW

**Comment by richardlford** [comment]
@agocke, please review.

File: src/Compilers/CSharp/CscCore/project.json [hunk 1]
```diff
@@ -1,42 +1,45 @@
 {
   "dependencies": {
-    "System.AppContext": "4.0.0-beta-22816",
```
Comment (agocke) [inline src/Compilers/CSharp/CscCore/project.json]: Can you also update the versions in the vbccore project.json

**Comment by jaredpar** [comment]
:+1: modulo updating the vbcore project.json file

File: src/Compilers/CSharp/CscCore/project.json (hunk 1 above)
Comment (richardlford) [inline src/Compilers/CSharp/CscCore/project.json]: The vbccore project.json was not the same as the csccore project.json. I consulted with @jaredpar and he said they should be the same and the C# version looked more correct. So I'll copy the C# version to the VB version.

Comment (richardlford) [inline src/Compilers/CSharp/CscCore/project.json]: I copied the csccore project.json to the vbccore project.json. I repushed (after rebasing and squashing).
I'll merge once testing passes.

**Comment by richardlford** [comment]
The testing has passed. I do not have write access to this repository. @jaredpar, @agocke, or someone else with write access, please merge. Thanks.

**Comment by richardlford** [comment]
@jaredpar, @agocke:

On further testing (with llilc) I discoved more stale ruleset files, namely:

src/Compilers/CSharp/CSharpCodeAnalysisRules.ruleset

(which is referenced when building Microsoft.CodeAnalysis.CSharp.dll)

So I decided to look at all the ruleset files. I discovered that 

src/Compilers/VisualBasic/BasicCodeAnalysisRules.ruleset

is also stale. In addition

build\TestProjectRules.ruleset

references a non-existent file (via non-existent directories):

..\src\Tools\Microsoft.CodeAnalysis.Toolset.Open\Rulesets\NonShippingProjectRules.ruleset

As soon as I’ve completed a successful bootstrap with LLILC I’ll push changes to these two files. 
I'll leave NonShippingProjectRules.ruleset as is, since I don't know what should be done with it.

**Comment by richardlford** [comment]
I've pushed fixes for the remaining stale ruleset files.

**Pull request merged**